builtin/vector.go

```diff
@@ -0,0 +1,33 @@
+package builtin
+
+import "github.com/spy16/slurp/core"
+
+var (
+	_ core.Any = (*Vector)(nil)
+	_ core.Seq = (*Vector)(nil)
```

Comment:
Logically `Vector` is a sequence. But in our Sabre discussions, we had decided to make it a `Seqable` (i.e., a type with `Seq() (Seq, error)` method) and define a `Vector` interface to allow for multiple Vector implementations. Any reasons why you went with Vector type which itself is a `Seq`? 

I am fine with this approach by the way (In Sabre model, I would've said no to this since special forms parser for example needed to differentiate a vector from a list to allow for `(let [...] ....)` kind of forms. But now that we have a swappable `Analyzer` and an intermediate `Expr` which doesn't care how the values got there, means we can use this approach. Might even be slightly more performance because no interface boxing around actual types)

Reply:
>Any reasons why you went with Vector type which itself is a Seq?

Honestly?  I had forgotten about our previous discussion, and I was in a rush to test something out... 😆 
I'll get this fixed up and ping you when it's ready.

>we had decided to make it a Seqable (

Can you remind me why we made this choice?  I gather from looking at https://github.com/spy16/sabre/pull/25 that the argument is basically the following:  lists are _defined_ by the `Seq` interface, so having `Vector` satisfy `Seq` would introduce some confusion.  For my own understanding, do you have an example of what this confusion would look like in practice?

Reply:
> Can you remind me why we made this choice?

Consider the `(fn [a b c] <expr>*)` special form for example. If the parser cannot differentiate between a list and a vector in some way, `(fn (a b c) <expr>*)` would be valid as well if we use the `Seq` type-assertion to make sure second argument is a vector, but the syntax we expect is the former. 

Since we were trying to support multiple implementations for a Vector (e.g., mutable vs immutable-persistent etc), this would have been a real problem in Sabre model because there was no intermediate `Expr` values and also no custom `Analyzer`.

Reply:
Ok that makes sense, thanks!